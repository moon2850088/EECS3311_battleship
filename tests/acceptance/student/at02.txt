--easy win
debug_test(easy)
fire([A,1])
fire([A,1])
bomb([A,1],[A,1])
bomb([A,1],[A,2])
bomb([A,3],[A,2])
bomb([C,4],[C,3])
fire([B,1])
fire([C,1])
fire([D,1])
fire([E,1])
fire([A,5])
fire([A,4])
fire([D,2])
fire([B,4])
