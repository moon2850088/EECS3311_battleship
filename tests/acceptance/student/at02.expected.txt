  state 0 OK -> Start a new game
->debug_test(easy)
  state 1 OK -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->fire([A, 1])
  state 2 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->fire([A, 1])
  state 3 Already fired there -> Keep Firing!
     1  2  3  4
  A  O  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->bomb([A, 1],[A, 1])
  state 4 Bomb coordinates must be adjacent -> Keep Firing!
     1  2  3  4
  A  O  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->bomb([A, 1],[A, 2])
  state 5 Already fired there -> Keep Firing!
     1  2  3  4
  A  O  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->bomb([A, 3],[A, 2])
  state 6 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  O  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 1/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->bomb([C, 4],[C, 3])
  state 7 OK -> 1x1 ship sunk! Keep Firing!
     1  2  3  4
  A  O  O  O  _
  B  _  _  _  v
  C  _  _  X  X
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->X
    1x1: [C, 3]->X
->fire([B, 1])
  state 8 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  O  _
  B  O  _  _  v
  C  _  _  X  X
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 2/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->X
    1x1: [C, 3]->X
->fire([C, 1])
  state 9 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  O  _
  B  O  _  _  v
  C  O  _  X  X
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 3/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->X
    1x1: [C, 3]->X
->fire([D, 1])
  state 10 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  O  _
  B  O  _  _  v
  C  O  _  X  X
  D  O  _  _  _
  Current Game (debug): 1
  Shots: 4/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->X
    1x1: [C, 3]->X
->fire([E, 1])
  state 11 Invalid coordinate -> Keep Firing!
     1  2  3  4
  A  O  O  O  _
  B  O  _  _  v
  C  O  _  X  X
  D  O  _  _  _
  Current Game (debug): 1
  Shots: 4/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->X
    1x1: [C, 3]->X
->fire([A, 5])
  state 12 Invalid coordinate -> Keep Firing!
     1  2  3  4
  A  O  O  O  _
  B  O  _  _  v
  C  O  _  X  X
  D  O  _  _  _
  Current Game (debug): 1
  Shots: 4/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->X
    1x1: [C, 3]->X
->fire([A, 4])
  state 13 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  O  O
  B  O  _  _  v
  C  O  _  X  X
  D  O  _  _  _
  Current Game (debug): 1
  Shots: 5/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->X
    1x1: [C, 3]->X
->fire([D, 2])
  state 14 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  O  O
  B  O  _  _  v
  C  O  _  X  X
  D  O  O  _  _
  Current Game (debug): 1
  Shots: 6/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->X
    1x1: [C, 3]->X
->fire([B, 4])
  state 15 OK -> 2x1 ship sunk! You Win!
     1  2  3  4
  A  O  O  O  O
  B  O  _  _  X
  C  O  _  X  X
  D  O  O  _  _
  Current Game (debug): 1
  Shots: 7/8
  Bombs: 2/2
  Score: 3/3 (Total: 3/3)
  Ships: 2/2
    2x1: [B, 4]->X;[C, 4]->X
    1x1: [C, 3]->X
